src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -518,15 +393,24 @@ internal bool RequiresMemberAccessForRead(SecurityException? securityException)
                 }
                 return true;
             }
-            if (this.BaseContract != null && this.BaseContract.RequiresMemberAccessForRead(securityException))
+            if (BaseClassContract != null && BaseClassContract.RequiresMemberAccessForRead(securityException))
                 return true;
```

Comment:
Although out of scope in this change, there's a lot of cleanup that can be done around this area as a lot of it is to accommodate CAS. securityException should never be non-null as it's CAS that would throw the exception during RefEmit.